polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.tsx

```diff
@@ -7,37 +7,59 @@ import {
   PositionedOverlay,
 } from '../../../PositionedOverlay';
 import {useI18n} from '../../../../utilities/i18n';
+import {getRectForNode} from '../../../../utilities/geometry';
+import type {TooltipPosition} from '../../Tooltip';
 
 import styles from './TooltipOverlay.scss';
 
 export interface TooltipOverlayProps {
   id: string;
   active: boolean;
-  preventInteraction?: PositionedOverlayProps['preventInteraction'];
   preferredPosition?: PositionedOverlayProps['preferredPosition'];
   children?: React.ReactNode;
   activator: HTMLElement;
   accessibilityLabel?: string;
   onClose(): void;
+  transform: TooltipPosition;
 }
 
 export function TooltipOverlay({
   active,
   activator,
   preferredPosition = 'below',
-  preventInteraction,
   id,
   children,
   accessibilityLabel,
+  transform,
 }: TooltipOverlayProps) {
   const i18n = useI18n();
+  const [tooltipTransform, setTooltipTransform] = useState<string>('');
+  const tooltipRef = useRef<HTMLDivElement>(null);
+  const windowSize = useRef({
```

Comment:
I think this is going to cause an error on any server-rendered pages, as the `window` object will not exist. Plus we need to account for when a user resizes their browser. It'd be safer to do something like:

```tsx
interface WindowSize {
  width: number;
  height: number; 
}
const windowSize = useRef<WindowSize>(null);
useEffect(() => {
  function handleWindowResize() {
    windowSize.current = {
      width: window.innerWidth,
      height: window.innerHeight
    }
  }
  handleWindowResize();
  window.addEventListener('resize', handleWindowResize);
  () => window.removeEventListener('resize', handleWindowResize);
}, [])
```